packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-field-auth-iam.ts

```diff
@@ -0,0 +1,634 @@
+import gql from 'graphql-tag';
+import { reconfigureAmplifyAPI, withTimeOut } from '../utils/api';
+import { GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
+import { Observable, ZenObservable } from 'zen-observable-ts';
+
+// to deal with bug in cognito-identity-js
+(global as any).fetch = require('node-fetch');
+
+// delay times
+const SUBSCRIPTION_DELAY = 10000;
+const SUBSCRIPTION_TIMEOUT = 10000;
+
+export const testRdsIAMFieldAuth = (engine: ImportedRDSType, queries: string[]): void => {
+  describe('RDS IAM field auth', () => {
+    const [db_user, db_password, db_identifier] = generator.generateMultiple(3);
+
+    // Generate settings for RDS instance
+    const username = db_user;
+    const password = db_password;
+    let region = 'us-east-1'; // This get overwritten in beforeAll
+    let port = getDefaultDatabasePort(engine);
+    const database = 'default_db';
+    let host = 'localhost';
+    const identifier = `integtest${db_identifier}`;
+    const engineSuffix = engine === ImportedRDSType.MYSQL ? 'mysql' : 'pg';
+    const engineName = engine === ImportedRDSType.MYSQL ? 'mysql' : 'postgres';
+    const projName = `${engineSuffix}modelauth2`;
+    const apiName = projName;
+
+    let projRoot;
+    let person1IAMPrivateClient, person2IAMPrivateClient, person3IAMPrivateClient;
+    let person1IAMPublicClient, person2IAMPublicClient, person3IAMPublicClient;
+    let apiEndPoint;
+
+    beforeAll(async () => {
+      projRoot = await createNewProjectDir(projName);
+      await initProjectAndImportSchema();
+      await amplifyPush(projRoot, false, {
+        useBetaSqlLayer: SQL_TESTS_USE_BETA,
+      });
+
+      // To handle a bug which doesn't populate cognito pool id in the generated resolvers on first push
+      await apiGqlCompile(projRoot, false, {
+        forceCompile: true,
+      });
+      await amplifyPush(projRoot, false, {
+        skipCodegen: true,
+        useBetaSqlLayer: SQL_TESTS_USE_BETA,
+      });
+
+      const meta = getProjectMeta(projRoot);
+      const appRegion = meta.providers.awscloudformation.Region;
+      const { output } = meta.api[apiName];
+      const { GraphQLAPIIdOutput, GraphQLAPIEndpointOutput, GraphQLAPIKeyOutput } = output;
+      apiEndPoint = GraphQLAPIEndpointOutput as string;
+
+      const { graphqlApi } = await getAppSyncApi(GraphQLAPIIdOutput, appRegion);
+
+      expect(GraphQLAPIIdOutput).toBeDefined();
+      expect(GraphQLAPIEndpointOutput).toBeDefined();
+      expect(GraphQLAPIKeyOutput).toBeDefined();
+
+      expect(graphqlApi).toBeDefined();
+      expect(graphqlApi.apiId).toEqual(GraphQLAPIIdOutput);
+
+      await createAppSyncClients(apiEndPoint, appRegion);
+    });
+
+    const createAppSyncClients = async (apiEndPoint, appRegion): Promise<void> => {
+      configureAmplify(projRoot);
+      const unAuthCredentials = await Auth.currentCredentials();
+      const [cognito_username, cognito_password] = ['test@test.com', 'Password123!'];
```

Comment:
very nit, but why snake case instead of camel case?

Reply:
It carried forward from our existing tests, I can correct it on the follow-up PR.